lxd/storage/backend_lxd.go

```diff
@@ -199,7 +207,7 @@ func (b *lxdBackend) Delete(localOnly bool, op *operations.Operation) error {
 	}
 
 	// Delete the low-level storage.
-	if !localOnly || !b.driver.Info().Remote {
+	if !localOnly || !b.driver.Info().Remote || !b.driver.Info().MountedRoot {
```

Comment:
@monstermunchkin there's quite a few "not" and "ors" here, I don't know about you, but I'm starting to struggle to reason about what this actually means. :)

Perhaps this is getting to the point of re-organising this to be instead just the conditions where we would do a b.driver.Unmount() and for all other conditions to the b.driver.Delete (after all it is a delete request, so the 'normal' route one would expect would be to call b.driver.Delete). The b.driver.Unmount call would make more sense as the exception here.

Reply:
@tomponline We could change that to
```go
if localOnly && b.driver.Info().Remote && b.driver.Info().MountedRoot {
	_, err := b.driver.Unmount()
	if err != nil {
		return err
	}
} else {
        // ...
}
```

Reply:
@monstermunchkin yeah I think that is much more readable. Thanks